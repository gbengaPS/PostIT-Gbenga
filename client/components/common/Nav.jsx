import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

/**
 * @param {object} props
 * @returns {object} -returns react element
 */
const Nav = (props) => {
  const logout = () => {
    localStorage.removeItem('postitUser');
    localStorage.removeItem('postitToken');
    props.history.push('/login');
  };
  // Check where postit logo links to.
  // It should link to dashboard if user is signed in, otherwise, landing page.
  let postitLink;
  if (localStorage.getItem('postitToken')) {
    postitLink = '/dashboard';
  } else {
    postitLink = '/';
  }
  return (
  <div className="row">
      <nav className="transparent col-s12">
        <ul id='userDropdown' className='dropdown-content'>
          <li>
            <a href="#" onClick={logout}>Logout</a>
          </li>
        </ul>
        <div className="nav-wrapper">
        <div>
          <div className="col s2 offset-s1">
          <Link to={postitLink} className="left brand-logo">
            Postit
          </Link>
          </div>
          <div className="col s2 ">
            <span className="center brand-logo">{props.middleLink}</span>
          </div>
          <div className="right">
            {props.rightLink}
          </div>
        </div>
        </div>
      </nav>
    </div>
  );
};
Nav.propTypes = {
  middleLink: PropTypes.string,
  rightLink: PropTypes.object,
  history: PropTypes.object,
};
export default Nav;
