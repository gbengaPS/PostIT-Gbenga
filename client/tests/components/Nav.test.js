import React from 'react';
import 'mock-local-storage';
import Nav from '../../components/common/Nav';

describe('Nav component', () => {
  it('should contain one nav tag', () => {
    const NavWrapper = shallow(<Nav />);
    expect(NavWrapper.find('nav').length).toBe(1);
  });
});
