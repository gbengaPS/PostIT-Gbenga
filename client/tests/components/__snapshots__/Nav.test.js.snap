// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component should render correctly 1`] = `
<div
  className="row"
>
  <nav
    className="transparent col-s12"
  >
    <ul
      className="dropdown-content"
      id="userDropdown"
    >
      <li>
        <a
          href="#"
          onClick={[Function]}
        >
          Logout
        </a>
      </li>
    </ul>
    <div
      className="nav-wrapper"
    >
      <div>
        <div
          className="col s2 offset-s1"
        >
          <Link
            className="left brand-logo"
            replace={false}
            to="/"
          >
            Postit
          </Link>
        </div>
        <div
          className="col s2 "
        >
          <span
            className="center brand-logo"
          />
        </div>
        <div
          className="right"
        />
      </div>
    </div>
  </nav>
</div>
`;
